### PR TITLE
Hiera integration for firewall rules

### PR DESCRIPTION
**Pull request by discreet**
I put in a class called custom_firewall.pp which allows you to create and remove your firewall rules using Hiera.

### CODE REVIEW

**Comment by daenney** [comment]
It is currently missing any form of documentation and tests, it's also causing the current tests to fail.

Is there any particular reason this wasn't added on the main class as a `$rules` parameter which is bound to undef or an empty array by default instead?

**Comment by elyscape** [comment]
Here's a better question: why is a personal `my_fw` implementation being submitted as a pull request to the module itself?

File: manifests/params.pp [hunk 1]
```diff
@@ -18,6 +19,10 @@
           }
           $service_name = 'iptables'
         }
+        'Gentoo': {
```
Comment (elyscape) [inline manifests/params.pp]: Why is Gentoo moved into the RedHat osfamily?  It's not a RedHat variant.

**Comment by jonnytpuppet** [comment]
I'm closing this as this has far too many features to be considered for a single pull request. You also need to add new tests for the new features added, and all current tests must still pass.

Also, commit should be squashed

Feel free to open a new pull request with updated commits.

**Pull request closed without merging**